client/src/app/models/user.js

```diff
@@ -29,7 +36,15 @@ window.User = Backbone.Model.extend({
         window.localStorage.setItem("_user.name", data.attributes.username);
         router.navigate('/', { trigger: true });
       },
-      error: function() { console.log("User signup error"); }
+      error: function() {
+        console.log("User signup error");
+        // _this.trigger("signupError", _this);
+        var options = {
+          text: "Error: Username or email is already taken. Please try again."
+        };
+        this.formErrorView = new formErrorView(options);
```

Comment:
In addition, ideally the error message should be the same as what the server sends back.